### PR TITLE
build: delete dead promQL compatibility test suite

### LINKED ISSUE #15294: Delete dead promql-e2e test suite
```
make test-promql-e2e
```

### PR DESCRIPTION
**Pull request by danxmoran**
Closes #15294 

This submodule doesn't build any more, and the tests cover behavior that was never officially released. @rbetts gave a 👍 in Slack to deleting the suite instead of fixing it up.

### CODE REVIEW

File: cmd/influxd/launcher/launcher.go [hunk 1]
```diff
@@ -574,7 +574,7 @@ func (m *Launcher) run(ctx context.Context, opts *InfluxdOpts) (err error) {
 
 	// NATS streaming server
 	natsOpts := nats.NewDefaultServerOptions()
-	natsOpts.Port = nats.RandomPort
+	natsOpts.Port = nats.RandomPort // FIXME: This is causing promql e2e failure, why does it work in general?
```
Comment (danxmoran) [inline cmd/influxd/launcher/launcher.go:577]: TODO: FIXME is fixed, remove the comment

**Comment by lesam** [comment]
PR title is a bit misleading.

**Comment by danxmoran** [comment]
> PR title is a bit misleading.

Good catch, leftover from the early draft that was trying to fix the tests. Updated!

**Comment by lesam** [comment]
Should we also be removing `github.com/influxdata/promql/v2` or marking it officially deprecated in some way ?

**Review by lesam — APPROVED** [review]
LGTM - though should we also delete the stuff being tested?

**Comment by lesam** [comment]
> Should we also be removing `github.com/influxdata/promql/v2` or marking it officially deprecated in some way ?

Also `github.com/influxdata/flux/promql` , `github.com/influxdata/query/promql` ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -260,9 +260,7 @@ jobs:
           name: parallel go race tests
           command: |
             set +e
-            # filter internal/promqltests because it has special go.mod
-            TESTFILES=$(go list ./... | grep -v internal/promqltests | circleci tests split --split-by=timings)
-            echo $TESTFILES
```
Comment (lesam) [inline .circleci/config.yml:265]: I like the `echo $TESTFILES` , should we keep it?

Comment (danxmoran) [reply .circleci/config.yml:265]: :+1: will restore it

**Pull request merged**